### PR TITLE
[MAINT] Refactor tags surface masker

### PR DESCRIPTION
**Pull request by Remi-Gau**
- Closes none

Changes proposed in this pull request:

- create a _BaseSurfaceMasker from which all surface maskers inherit
- set the tags on the base masker

### CODE REVIEW

**Comment by Remi-Gau** [comment]
@man-shu does this refactoring make sense to you?

**Review by man-shu — APPROVED** [review]
Yes! Very clear indeed

File: nilearn/maskers/base_masker.py [hunk 1]
```diff
@@ -396,3 +396,33 @@ def _check_fitted(self):
                 "has not been fitted. "
                 "You must call fit() before calling transform()."
             )
+
+
+class _BaseSurfaceMasker(TransformerMixin, CacheMixin, BaseEstimator):
+    """Class from which all surface maskers should inherit."""
```
Comment (man-shu) [inline nilearn/maskers/base_masker.py:402]: Maybe a "version added" tag? But I guess it doesn't really matter because it wont be public anyway

Comment (bthirion) [reply nilearn/maskers/base_masker.py:402]: +addition in Whatsnew

Comment (Remi-Gau) [reply nilearn/maskers/base_masker.py:402]: hmmm not sure given this is a private class
I'd wait before adverstising it

**Pull request merged**